### ISSUE TITLE
Consolidate axis `bandPosition` and `tickExtra` 

### ISSUE THREAD

**Issue by kanitw** [vega/vega-lite#5443]
The current usage of `bandPosition` and `tickExtra` are unnecessarily complex.

The only sensible way to use these two properties are to set bandPosition to `1` and `tickExtra` to `true`. 

```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v4.json",
  "description": "A simple bar chart with embedded data.",
  "data": {
    "values": [
      {"a": "A", "b": 28}, {"a": "B", "b": 55}, {"a": "C", "b": 43},
      {"a": "D", "b": 91}, {"a": "E", "b": 81}, {"a": "F", "b": 53},
      {"a": "G", "b": 19}, {"a": "H", "b": 87}, {"a": "I", "b": 52}
    ]
  },
  "mark": "bar",
  "encoding": {
    "x": {"field": "a", "type": "ordinal", "axis": {
      "bandPosition": 1,
      "tickExtra": true
    }},
    "y": {"field": "b", "type": "quantitative"}
  }
}
```

Setting bandPosition to `0` and `tickExtra` to `true` doesn't work as tickExtra always add a tick at the start of the axis.   

I don't see why one would ever set `bandPosition` to a value other than 1 (and 0.5, but 0.5 is already the default).  

I think we should simplify this by hiding `tickExtra` and make `bandPosition` accepts either `"center"` or `"side"` (which automatically enables "tickExtra").  We can still take number for backward compatibility reason, but I think it's an overkill, esp. for Vega-Lite users. 

Unless we have a convincing use case for bandPosition != 1 and != 0.5, I strongly we should do this simplication in Vega-Lite (as a part of 4.0).

(We might even wanna do this in Vega, but I don't have as strong opinion for Vega.) 

@domoritz  @jheer  -- any thoughts?

cc: @haldenl

**Comment by domoritz** [comment]
I agree that tickExtra should really only be on if bandPosition is > 0 and most sense for bandPosition = 1. 

I'm not convinced by using strings for bandPositon. Would you be okay with just leaving it at what it is? If you really want a string, we should probably do `left`, `center`, and `right` like for align but that will mean that we need separate ops for x and y. 

I'm not yet convinced that this is such a major issue and it's worth diverging from Vega.

**Comment by kanitw** [comment]
Ok, maybe I didn't explain my rationale well enough.  

For context, yesterday @haldenl and I can't figure this out for a while.  
(If it confuses us, it will confuse most users.)


We tried to use `bandPosition` and recognize that a tick is missing.  We then thought this was a bug.   I only remembered that there is `tickExtra` property an hour later.  I then thought about adding reference to `tickExtra` in the `bandPosition` docs so people will discover it. 
However, thinking more, it seems an overkill that users need to discover the `tickExtra` property because user intention would be either having the tick in the middle or on the sides (which requires one extra tick).  

If there are only two possible user goals, we shouldn't require two different properties.  

If you still disagree, I'd be curious to hear reasons whys.


(Btw, using `left`, `center`, and `right` is totally missing the point since tickExtra doesn't work with left.  Plus, user intention won't really be to place the tick on the left, but rather on the sides of the bands.)

**Comment by domoritz** [comment]
I see. This helps me understand the request a bit better. I didn't realize this was about discoverability.

I'm still not convinced by the values you want, though. center and side are not terms we use otherwise. Maybe start, center/middle, end or something like it would be more consistent. What properties do we have that control similar things? What terminology do they use?

**Comment by kanitw** [comment]
Well, start and end definitely doesn't make sense. Unlike title anchors that we may want to be on one of the sides (either "start" or "end"), you want the ticks to be on _both_ sides for each band.  

![image](https://user-images.githubusercontent.com/111269/66272346-7cfa8800-e81d-11e9-9f4d-bbdc094e3ea0.png)


I'm open to a new keyword suggestions, but there should be only two options (not 3): "center"/"middle" or "sides"/"side"/"end".

If we want similar term to title anchor, I can see "middle" and "end" as a reasonable option, but then it won't be clear that it behaves slightly different from anchor. 

(I don't have a strong opinion about the keywords here, but I think having both `bandPosition: number` and `tickExtra: boolean` is unnecessary complicated and confused even myself.)

**Comment by domoritz** [comment]
To not break backwards compatibility and maintain consistency with Vega, would you be okay with making `tickExtra=true` by default if `bandPosition=1`? That would solve the discoverability issue but still give users flexibility.

Otherwise, I would push for changing Vega so that we remain consistent.

**Comment by kanitw** [comment]
> To not break backwards compatibility and maintain consistency with Vega, would you be okay with making `tickExtra=true` by default if `bandPosition=1`?

I thought about that, but that is also not great because it's unclear why bandPosition=1 would work but bandPosition=0 would not. 

Again, the user intention would be putting ticks on the sides of the bands.  Having to set bandPosition=1 (and tickExtra, if it is not set automatically) introduces a gulf of execution.  

FWIW, I also played with tickExtra and thought it was buggy until I read that docs and found that it somehow only works with  bandPosition=1.  I'd argue that we shouldn't require users to read the docs to discover this. 

> I would push for changing Vega so that we remain consistent. 

Probably. 

> To not break backwards compatibility

We could consider adding `bandTickPosition/tickPosition: 'middle' | 'side(s)'` and hiding/deprecating `tickExtra` and `bandPosition`. 

(After all, it's a position of the tick anyway.)

**Comment by domoritz** [comment]
Paging @arvind and @jheer for their opinions on this.

**Comment by kanitw** [comment]
Note that I figure we better do this before we do 4.0.
